doc/generic/pgf/CHANGELOG.md

```diff
@@ -16,6 +16,13 @@ lot of contributed changes. Thanks to everyone who volunteered their time!
   `\pgfrevision{,date}` and `\pgfversion{,date}` are synonyms for now, but the
   revision should eventually gain back its original meaning.  However, as of now
   this is not supported by l3build.
+- Many operations in `pgfkeys` used to use `\csname` directly which lets the
+  given csname become `\relax` in case it wasn't defined. This resulted in some
+  leakage into of accidentally `\relax`ed keys into the global scope. This has
+  been cleaned up a little. To preserve compatibility macros that used to expand
+  to a `\relax`ed csname now expand to a primitive `\relax`. However, this means
+  that the result of these macros must not be assigned to since this would
+  overwrite the primitve `\relax`.
```

Comment:
It seems `\pgfkeysvalueof` and `\pgfkeysgetvalue` are the only two directly affected. Mention them here?

On line 21, is the first `into` after `leakage` redundant?

It takes me some time to understand the last two sentences, hence might take other readers longer. I understand a changelog entry should use only text descriptions whenever possible, but still ... . Maybe leave a comment containing an explanatory example under this PR and append the (unfortunately long) link to the changelog entry?

feel I've made too many comments (than acceptable) here 😞

Reply:
For the last sentence, maybe the following is easier to understand: "However, this means that you must not make assignments to the result of these macros' expansion since this would overwrite the primitive `\relax`".

Also for that only `\pgfkeysvalueof` is really affected, `\pgfkeysgetvalue` doesn't yield `\relax` but lets `\<user-provided-cs>` to `\relax` (huge difference) assigning to which is fine. And assigning to the result of `\pgfkeysvalueof` seems strange anyways (while I could imagine use cases I don't think those are common, feel free to correct me here).

However: I wonder whether that's the best behaviour anyway (I know, too late to change, but I'd prefer it threw an expandable error that the key isn't defined and return just nothing, and yes, that could be build in two steps of expansion).

Reply:
> Also for that only `\pgfkeysvalueof` is really affected

@Skillmon You're right. Of course that includes both direct and indirect uses.